packages/@lwc/ssr-compiler/src/compile-template/transformers/if.ts

```diff
@@ -63,7 +63,7 @@ function bIfStatement(
 
     return b.ifStatement(
         expressionIrToEs(condition, cxt),
-        bBlockStatement(children, cxt, true),
+        bBlockStatement(children, cxt, !cxt.isSlotted),
```

Comment:
This removes 3 expected failures related to `lwc:if`, but the real solution for it in light-dom which would fix `slot-not-at-top-level/advanced/lwcIf/light/index.js` is a bit more complicated.

Reply:
We might want to only do this if the `lwc:if` is at the top level of the slotted content. 

Looking at the [way](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/rendering.ts#L718) we flatten slotted content, we only do it for the top-level elements being slotted.

For example:

```html
<template lwc:render-mode="light">
    <x-slot>
        <h1 slot="top">top content</h1>
        <h1 slot="bottom">
            <template lwc:if={isTrue}>
                <span>rendered</span>
            </template>
        </h1>
    </x-slot>
</template>
```

```html
<template lwc:render-mode="light">
    <x-leaf>
        <slot name="top" slot="bottom"></slot>
        <slot name="bottom" slot="top"></slot>
    </x-leaf>
</template>
// x-slot
```

```html
<template lwc:render-mode="light">
    <slot name="top"></slot>
    <slot name="bottom"></slot>
</template>
// x-leaf
```

The bookends around the `lwc:if` will be removed:

```html
  <x-light-container>
    <x-slot>
      <x-leaf>
        <!---->
        <h1>
-         <!---->
          <span>
            rendered
          </span>
-         <!---->
        </h1>
        <!---->
        <!---->
        <h1>
          top content
...
```

We could probably do this in a follow-up PR though.